File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int left = 0,right = 0;
        int ans = INT_MAX,n=nums.size(),sum = 0;
        while(right<n){
               sum += nums[right++];
            while(sum >= target){
                ans = min(ans,right-left);
                sum -= nums[left++];
            }
        }
        return ans == INT_MAX ? 0: ans;  
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left, right = 0, 0
        ans, n, arrSum = float('inf'), len(nums), 0 
        while right < n:
            arrSum += nums[right]
            right += 1
            while arrSum >= target:
                ans = min(ans,right-left)
                arrSum -= nums[left]
                left += 1
        return 0 if ans == float('inf') else ans